src/decorators/headerFooter.js

```diff
@@ -16,30 +17,25 @@ function headerFooter(ComposedComponent) {
         collapsed: props.collapsible && props.collapsible.collapsed
       };
 
-      this._headerElement = this._getElement({
-        type: 'header',
-        handleClick: props.collapsible ? this.handleHeaderClick : null
-      });
-
       this._cssClasses = {
         root: cx('ais-root', this.props.cssClasses.root),
         body: cx('ais-body', this.props.cssClasses.body)
       };
 
       this._footerElement = this._getElement({type: 'footer'});
     }
-    shouldComponentUpdate(nextProps, nextState) {
-      return nextState.collapsed === false ||
```

Comment:
Maybe we need a `Header` and `Footer` components, with their own `shouldComponentUpdate` methods?

I feel like our `shouldComponentUpdate`s are too weak at the moment and are not worth the complexity cost. Does it really bring any perf improvement? I don't really know that part but from reading [this doc](https://facebook.github.io/react/docs/advanced-performance.html) it seems that it only really make sense with immutability in mind (which we only have in the helper).